### DISCUSSION TITLE
How to convert multiple piped files into a single SafeTensor file?

### DISCUSSION THREAD

**Discussion by xxddccaa** [huggingface/diffusers#7336] [General]
How to convert multiple piped files into a single SafeTensor file?

For example, from this address: https://huggingface.co/Vargol/sdxl-lightning-4-steps/tree/main

```python
import torch
from diffusers import StableDiffusionXLPipeline, EulerDiscreteScheduler

base = "Vargol/sdxl-lightning-4-steps"

pipe = StableDiffusionXLPipeline.from_pretrained(base, torch_dtype=torch.float16).to("cuda")
```

How can I convert `pipe` into a single SafeTensor file as a whole?

**Comment by xxddccaa** [comment]
Just like the file `sd_xl_base_1.0_0.9vae.safetensors`, which contains the components needed from `diffusers`.

**Comment by asomoza** [comment]
You can use the script: https://github.com/huggingface/diffusers/blob/main/scripts/convert_diffusers_to_original_sdxl.py

for example:

```bash
 python convert_diffusers_to_original_sdxl.py --model_path "models/sdxl-lightning-4-steps" --checkpoint_path "models/vargol-lighting.safetensors" --use_safetensors
```

**Comment by xxddccaa** [comment]
Wow, you're really awesome! I've seen a lot of conversions along this path, and they're all so cool. I'm gonna give it a shot right now.

**Comment by xxddccaa** [comment]
this method works  fine. Thanks!

**Comment by xxddccaa** [comment]
```
import time

import torch
from diffusers import StableDiffusionXLPipeline, EulerDiscreteScheduler

base = "/data/xiedong/diffusers/scripts/vargol-lighting.safetensors"

pipe = StableDiffusionXLPipeline.from_single_file(base, torch_dtype=torch.float16).to("cuda")

# Ensure sampler uses "trailing" timesteps.
pipe.scheduler = EulerDiscreteScheduler.from_config(pipe.scheduler.config, timestep_spacing="trailing")

time1 = time.time()
# Ensure using the same inference steps as the loaded model and CFG set to 0.
img = pipe("A girl smiling", num_inference_steps=4, guidance_scale=0).images[0]
time2 = time.time()
img.save("output1.png")
print(time2 - time1)

```

**Reply by feiyuhuahuo** [reply]
Hi， nice example! Can you show some code of  directly loading .safetensors file of ControlNet ? I'm really confused of this.

**Reply by asomoza** [reply]
Hi, are you using a specific controlnet? Most of the controlnets for SDXL are in the diffusers format so they don't need to be loaded with `from_single_file`.
